tango/integrations/beaker/executor.py

```diff
@@ -533,16 +542,23 @@ def future_done_callback(future: concurrent.futures.Future):
                             result_location=self.workspace.step_info(step).result_location,
                             logs_location=future.result(),
                         )
+                        steps_left_to_run.discard(step)
+                    elif isinstance(exc, ResourceAssignmentError):
+                        submitted_steps.discard(step_name)
```

Comment:
This doesn't mean they are discarded forever, does it?

Reply:
Nope, they get put back into the queue, or I guess _kept_ in the queue. The important queue here is `steps_left_to_run`.